presto-main/src/test/java/io/prestosql/execution/buffer/TestBroadcastOutputBuffer.java

```diff
@@ -316,6 +317,40 @@ public void testSharedBufferFull()
         enqueuePage(buffer, createPage(3));
     }
 
+    @Test
+    public void testNotifyStatusOnBufferFull()
+    {
+        AtomicInteger notifyCount = new AtomicInteger();
+        BroadcastOutputBuffer buffer = createBroadcastBuffer(
+                createInitialEmptyOutputBuffers(BROADCAST).withBuffer(FIRST, BROADCAST_PARTITION_ID),
+                sizeOfPages(1),
+                notifyCount::incrementAndGet);
+
+        // Add a page to the buffer
+        addPage(buffer, createPage(1));
+        assertTrue(buffer.isFull().isDone());
+        assertEquals(notifyCount.get(), 0);
+
+        // Add another page to block
+        ListenableFuture<?> future = enqueuePage(buffer, createPage(2));
+        assertFalse(future.isDone());
+        assertEquals(notifyCount.get(), 1);
+
+        // Set no more buffers
```

Comment:
Please add another test case where you make sure notification count is still `0` when no more buffers state is reached, but buffer is full

Reply:
done